src/query.rs

```diff
@@ -1269,6 +1269,9 @@ pub struct View<'q, Q: Query> {
     fetch: Vec<Option<Q::Fetch>>,
 }
 
+unsafe impl<Q: Query> Send for View<'_, Q> {}
```

Comment:
I think both impls need to be conditional on
```rust
<Q::Fetch as Fetch<'q>>::Item: Send
```
as otherwise values `T: !Sync` could be accessed from multiple threads if e.g. `Item == &'q T`.

Reply:
I think it's ok because components need to be Send + Sync

https://github.com/Ralith/hecs/blob/master/src/world.rs#L1025

A lot of code would be broken otherwise :D

Reply:
Updated the other bounds too

Reply:
> I think it's ok because components need to be Send + Sync

Right, I forgot about that. I do prefer the explicit bounds here though to enable "local reasoning."

> Updated the other bounds too

I think `Item: Send` should suffice for `View: Sync` etc. as access to `&View` will always yield `Item`, never `&Item`.

Reply:
Would it make more sense to constrain `Item` (and perhaps `State`) in the definition of `Fetch`?

Reply:
> Would it make more sense to constrain Item (and perhaps State) in the definition of Fetch?

I don't there is anything in `Fetch` that requires `Send` or `Sync`? Everything in `mod query` can work without it, except when queries etc. need to parallelized or sent to/accessed from other threads.

IMHO, I think either the bounds proposed here are preferable due to explicitness or relying on the bounds on `Component` is preferable due to convenience.